### PR TITLE
Add recipe for umi-transfer

### PR DESCRIPTION
**Pull request by MatthiasZepper**
This PR adds a recipe for [`umi-transfer`](https://github.com/SciLifeLab/umi-transfer), a tool for transferring Unique Molecular Identifiers (UMIs) from a separate FastQ file into the header of reads, while ensuring by ID comparison that only correct UMIs are integrated. This functionality was previously contained in [barcodex-rs](https://github.com/oicr-gsi/barcodex-rs), which is unfortunately no longer maintained and thus fails to build on Bioconda. 

The new tool is developed by scientists at the [Swedish National Genomics Infrastructure](https://ngisweden.scilifelab.se/) and published through the [SciLifeLab organisation](https://github.com/SciLifeLab/umi-transfer). The first properly tested version 1.0.0 "_Ardent Adenine_" was released on 2023-08-16.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by bgruening** [comment]
Is this still in Draft?

**Comment by MatthiasZepper** [comment]
> Is this still in Draft?

Yes. Thank you very much for double-checking, though! I will make use of the bot when ready! 

The reason for it is, that it points to my personal fork of the software. But I[ can not make a release on the official repo without approval by a second person within the organization](https://github.com/SciLifeLab/umi-transfer/pull/10), and at the moment literally everyone is on holidays. I just wanted to make sure that tests pass, so I can tick off "Adding to Bioconda" on my ToDo list more or less.

**Comment by MatthiasZepper** [comment]
@BiocondaBot please fetch artifacts

**Comment by MatthiasZepper** [comment]
@BiocondaBot please fetch artifacts

**Comment by MatthiasZepper** [comment]
@BiocondaBot please update
@BiocondaBot please add label

**Comment by MatthiasZepper** [comment]
@BiocondaBot please add label

File: recipes/umi-transfer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+test:
+  commands:
+    - umi-transfer --help
+
+about:
+  home: https://github.com/oicr-gsi/barcodex-rs
+  license: MIT
+  license_family: MIT
```
Comment (mencian) [inline recipes/umi-transfer/meta.yaml:36]:
```suggestion
  license_family: MIT
  license_file: LICENSE
```

**Review by mencian — CHANGES_REQUESTED** [review]
Please add the license_file packaged with source, otherwise looks good to me!

**Review by mencian — APPROVED** [review]

**Pull request merged**